config/pipeline.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use Zend\Expressive\Helper\ServerUrlMiddleware;
+use Zend\Expressive\Helper\UrlHelperMiddleware;
+use Zend\Expressive\Middleware\NotFoundHandler;
+use Zend\Stratigility\Middleware\ErrorHandler;
+
+/**
+ * Setup middleware pipeline:
+ */
+
+// The error handler should be the first (most outer) middleware to catch all Exceptions.
+$app->pipe(ErrorHandler::class);
+$app->pipe(ServerUrlMiddleware::class);
+
+// Pipe more middleware here that you want to execute on every request:
+// - bootstrapping
+// - pre-conditions
+// - modifications to outgoing responses
+//
+// Piped Middleware may be either callables or service names. Middleware may also be passed as an array; each item in
+// the array must resolve to middleware eventually (i.e., callable or service name).
+//
+// Middleware can be attached to specific paths, allowing you to mix and match applications under a common domain.
+// The handlers in each middleware attached this way will see a URI with that PATH SEGMENT STRIPPED !!!
+// - $app->pipe('/api', $apiMiddleware);
+// - $app->pipe('/docs', $apiDocMiddleware);
+// - $app->pipe('/files', $filesMiddleware);
+
+// Register the routing middleware in the middleware pipeline
+$app->pipeRoutingMiddleware();
+$app->pipe(UrlHelperMiddleware::class);
+
```

Comment:
Now that zendframework/zend-expressive#413 is merged, we should also add in the `ImplicitHeadMiddleware` and `ImplicitOptionsMiddleware` to the pipeline, immediately following the routing middleware.